### ISSUE TITLE
Unable to resolve path to module 'export-to-csv'

### ISSUE THREAD

**Issue by markhats** [alexcaza/export-to-csv#108]
When trying to import export-to-csv into a js create-react-app via
```
import { mkConfig, generateCsv, download } from 'export-to-csv';
```
I'm getting the following linter warning. 
```
Unable to resolve path to module 'export-to-csv'
```
I then get a similar error when building the production version of the app and it fails to build.

Any ideas what might be causing this?

Thanks.

**Comment by alexcaza** [comment]
Just commenting to say I see this, and I'll attempt a reduced test case when I can. Live has been more busy lately :)

In the meantime, If you could replicate it in a stackblitz, or a github repo I could clone, to make the process quicker, that would be excellent! 😄

**Comment by markhats** [comment]
Sorry - I haven't had a chance to create a simple test case that exhibits the issue. 

It seems that it is actually really just a linter error and can be disabled using the following on the line above the import:
```
// eslint-disable-next-line import/no-unresolved
import { mkConfig, generateCsv, download } from 'export-to-csv';
``` 
I've no idea why it is only the `export-to-csv` import causing this though.

**Comment by alexcaza** [comment]
If that works, and you're comfortable running with that, then great 😄 If you want further help, a basic repo with a stripped-down version of your project would be great! I suspect it's a project config-related issue, though.

I'll leave this issue open for a while longer (until mid-July) if you have time to come back to it.

For clarity's sake, a stripped-down repo would contain a `package.json` with versions locked to the same versions in your main project, config files set up as you have them in your main project and only the bare components or functionality you're trying to test!

**Comment by ajanauskas** [comment]
I also encountered same issue

**Comment by alexcaza** [comment]
> I also encountered same issue

@ajanauskas would you have a reduced test case you could share? Like I said above I'm not able to replicate, and I suspect it's a config issue but I can't confirm that without more details 😄

**Comment by ajanauskas** [comment]
would be happy to if I understood whats causing this. My codebase is pretty big, lots of dependencies with most of them being private so its not that easy to replicate it

**Comment by Yousif-FJ** [comment]
I fixed the issue by updating typescript version in my project.

**Comment by alexcaza** [comment]
We've hit the mid-July mark for this issue. I'm going to close it for now since no one seems to be able to give a reduced test case for it. Feel free to open a new issue with one if it doesn't get resolved.

**Comment by ZengLawrence** [comment]
I had same issue. Was able to fix by upgrading typescript to 5.6.3. 

Looked like it was caused by transcomplie issue rather than support for ESM.

Lastly, [this gist](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) posted above really helped explain all error scenarios.

@alexcaza, thank you for maintaining this package.
